packages/lib-react-components/src/Markdownz/Markdownz.js

```diff
@@ -1,34 +1,146 @@
+  }
+
+  // Support image resizing, video, and audio using markdown's image syntax
+  renderMedia(nodeProps) {
+    let altText, width, height
+    const imgSizeRegex = /=(\d+(%|px|em|rem|vw)?)x(\d+(%|px|em|rem|vh)?)/
+    let alt = nodeProps.alt
+    const src = nodeProps.src
+    const match = alt.match(imgSizeRegex)
+
+    if (match && match.length > 0) {
+      width = match[1]
+      height = match[3]
+      altText = alt.split(match[0])[0].trim()
+    }
+    const mimeType = mime.lookup(src)
+
+    if (mimeType && mimeType.includes('video')) return <Video a11yTitle={alt} controls="below" preload='metadata' src={src} />
+    if (mimeType && mimeType.includes('audio')) return <audio controls preload='metadata' src={src} /> // Grommet doesn't have an audio component
+
+    return <Image alt={altText || alt} src={src} width={width} height={height} />
   }
-  const sanitizedMarkdownContent = sanitizeForMarkdown(children)
-
-  // Grommet Markdown is not styling anchors correctly to theme
-  // Issue: https://github.com/grommet/grommet/issues/2515
-  return (
-    <GrommetMarkdown options={mergedOptions}>
-      {sanitizedMarkdownContent}
-    </GrommetMarkdown>
-  )
+
+  render() {
+    const { children, components, settings } = this.props
+
+    if (!children) return null
+
+    if (Object.keys(components).includes('img')) {
+      console.warn('Overriding the rendering function for the img tag may break the syntax support for image resizing and using image markup for video and audio. Are you sure you want to do this?')
+    }
+
+    const componentMappings = {
+      a: Anchor,
+      h1: (nodeProps) => <Heading level='1'>{nodeProps.children}</Heading>,
+      h2: (nodeProps) => <Heading level='2'>{nodeProps.children}</Heading>,
+      h3: (nodeProps) => <Heading level='3'>{nodeProps.children}</Heading>,
+      h4: (nodeProps) => <Heading level='4'>{nodeProps.children}</Heading>,
+      h5: (nodeProps) => <Heading level='5'>{nodeProps.children}</Heading>,
+      h6: (nodeProps) => <Heading level='6'>{nodeProps.children}</Heading>,
+      img: (nodeProps) => this.renderMedia(nodeProps),
+      p: Paragraph,
+      span: Text,
+      table: Table,
+      tfoot: TableFooter,
+      thead: TableHeader,
+      tbody: TableBody,
+      td: TableCell,
+      tr: TableRow
+    }
+
+    const remarkReactComponents = Object.assign({},  componentMappings, components)
+    const remarkSettings = Object.assign({}, { footnotes: true }, settings)
+
+    const markdown = remark()
+      .data('settings', remarkSettings)
+      .use(emoji)
+      .use(remarkSubSuper)
+      .use(externalLinks)
+      .use(ping, {
+        ping: (resource, symbol) => this.shouldResourceBeLinkable(resource, symbol), // We could support passing in a prop to call a function here
+        pingSymbols: [at, hashtag, subjectSymbol],
+        resourceURL: (resource, symbol) => this.buildResourceURL(resource, symbol),
+        matchRegex: /(@[\w]+)|(^#[\w]+$)|(^\^S[0-9]+$)/
```

Comment:
Maybe use `pingSymbols` to build this expression, so that you don't have to edit symbols in two places in the code.

Reply:
`@[\w]+` will miss valid usernames that have non-alphabetic characters in them.

Reply:
I'm working on this, but bizarrely updating this is breaking other remark plugins; specifically the remark-emoji plugin. I think what this is is actually a bug in the ping plugin itself returning an undefined value for a hast node and then the emoji plugin is breaking because it isn't doing an existential check on the nodes. I'm going to try to fix this. The use of the regex in the  ping plugin seems to be brittle, but I'm not sure how else to do this. The original plugin used regex as well as our custom plugin with markdown-it.

Reply:
The issue isn't the regex for the username itself, it's the check for the preceding space: `(^|\s)`. Removing that, the parser runs fine. Adding it, then remark-emoji breaks, I think because with it remark-ping is now returning something as undefined when it shouldn't.

Reply:
I don't think this needs to test for a leading space. I think the regex should be
`matchRegex: /(@[\w\-\.]+)|(^#[\w]+$)|(^\^S[0-9]+$)/`

The pattern seems to be the match to apply to each token, after the tokenizer has done its thing.

Reply:
We could test that with the existing regex by running a test for `sometext@srallen` and seeing if that skips your username when the preceding space isn't present.

Reply:
I need to double check, but I think it'll miss the username mention if in the middle of a sentence without checking for a space.

Reply:
Add that to the tests first, then fix them if they break.

Reply:
Looking at the original ping code, they don't search for leading space either.
https://github.com/zestedesavoir/zmarkdown/blob/44a120b1d67013b2d19f017448877490e5615a1f/packages/remark-ping/src/index.js#L9

I could be wrong but I think the regex is matched against parsed tokens, not the full markdown text.

Reply:
Unfortunately I can't use tests to help fix this since the tests won't run because the parser is throwing.

Reply:
Actually I can write tests with the check for the space removed from the regex to make sure it's correctly parsing for at-mentions.